### PR TITLE
NIFI-9255: Support asynchronous session commits in ListenHTTP

### PR DESCRIPTION
**Pull request by turcsanyip**
https://issues.apache.org/jira/browse/NIFI-9255

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by Lehel44 — APPROVED** [review]

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java [hunk 1]
```diff
@@ -413,12 +414,20 @@ protected void proceedFlow(final HttpServletRequest request, final HttpServletRe
...
 
             session.transfer(flowFileSet, ListenHTTP.RELATIONSHIP_SUCCESS);
-            session.commitAsync();
+
+            final AsyncContext asyncContext = request.startAsync();
+            session.commitAsync(() -> {
+                        response.setStatus(this.returnCode);
+                        asyncContext.complete();
+                    }, t -> {
+                        response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:427]: @turcsanyip What do you think about logging the exception in this scenario?

Comment (turcsanyip) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:429]: @exceptionfactory Added the error logging. Thanks!

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java [hunk 2]
```diff
@@ -424,6 +424,8 @@ protected void proceedFlow(final HttpServletRequest request, final HttpServletRe
                         response.setStatus(this.returnCode);
                         asyncContext.complete();
                     }, t -> {
+                        logger.error("Failed to commit session. Returning error response to Remote Host: [{}] Port [{}] SubjectDN [{}] IssuerDN [{}]",
+                                new Object[] {request.getRemoteHost(), request.getRemotePort(), foundSubject, foundIssuer}, t);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:428]: Thanks for adding the log message! The wrapping `Object[]` can be removed:
```suggestion
                                request.getRemoteHost(), request.getRemotePort(), foundSubject, foundIssuer, t);
```

Comment (Lehel44) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:428]: This object wrapping is very common due to legacy usage. Do you think we could deprecate `ComponentLog::void error(String msg, Object[] os, Throwable t)` ?

Comment (turcsanyip) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java]: @exceptionfactory I applied your suggestion. Also for the other occurrences in this class.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @turcsanyip, looks good!

**Review by Lehel44 — APPROVED** [review]

**Comment by markap14** [comment]
Thanks @turcsanyip looks good to me too. Merged to main.

**Pull request merged**